### ISSUE TITLE
bug: Failed to upload file with spaces

### ISSUE THREAD

**Issue by junnplus** [datafuselabs/databend#6139]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

main

### What's Wrong?
```
object error: (op: write, path: tnc7yee14/stage/test/she et.csv, source: build request https://s3/tnc7yee14/stage/test/she et.csv: http::Error(InvalidUri(InvalidUriChar)))
```
### How to Reproduce?

Upload a file with spaces via curl `/v1/upload_to_stage`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by Xuanwo** [comment]
Related to https://github.com/Xuanwo/reqsign/pull/80

**Comment by Xuanwo** [comment]
Should be fixed by https://github.com/datafuselabs/opendal/pull/398 which will be released as opendal v0.9.1
